File: frontend/src/types/graph.ts
export interface GraphNode {
  id: string;
  entity_type: 'observable' | 'threat_actor' | 'campaign' | 'technique';
  label: string;
  metadata: Record<string, unknown>;
}

export interface GraphEdge {
  source: string;
  target: string;
  relationship_type: string;
  metadata: Record<string, unknown>;
}

export interface GraphData {
  nodes: GraphNode[];
  edges: GraphEdge[];
}

File: src/cti/schemas/graph.py
from pydantic import BaseModel


class GraphNode(BaseModel):
    id: str
    entity_type: str  # "observable" | "threat_actor" | "campaign" | "technique"
    label: str
    metadata: dict = {}


class GraphEdge(BaseModel):
    source: str
    target: str
    relationship_type: str
    metadata: dict = {}


class GraphData(BaseModel):
    nodes: list[GraphNode]
    edges: list[GraphEdge]
